### PR TITLE
Check new peripheral names against regex

### PR DESCRIPTION
**Pull request by aakashnshah**
Check new peripheral names against regex

### CODE REVIEW

File: dawn/js/components/peripherals/NameEdit.js [hunk 1]
```diff
@@ -8,10 +8,13 @@ var NameEdit = React.createClass({
     id: React.PropTypes.string
   },
   dataChange(data) {
-    Ansible.sendMessage('custom_names', {
-      id: this.props.id,
-      name: data.name
-    });
+    var x = new RegExp("^[A-Za-z][A-Za-z0-9]+$");
+    if (x.test(data.name)) {
```
Comment (AllanYangZhou) [inline dawn/js/components/peripherals/NameEdit.js]: I suggest you do the checking in a separate method, called checkNameValid.

**Comment by nikitakit** [comment]
You may want ```^[A-Za-z][A-Za-z0-9]*$```, not ```^[A-Za-z][A-Za-z0-9]+$```

**Comment by AllanYangZhou** [comment]
^ Yes, that should prevent a number from being at the start, which you guys were talking about earlier.

Also tagging @nfong1996 @mud13

**Comment by AllanYangZhou** [comment]
Also, I think you will want to alert the user that the name is invalid. Look at uses of smalltalk throughout this project to make this work.

**Pull request merged**